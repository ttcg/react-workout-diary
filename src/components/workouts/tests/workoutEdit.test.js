import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WorkoutEdit from '../workoutEdit';

it('renders without crashing', () => {
    const
        onChange = jest.fn(),
        onChangeDate = jest.fn(),
        onEdit = jest.fn(),
        toggle = jest.fn(),
        maxDate = new Date('2019-01-31'),
        item = { date: new Date('2019-01-02') };

    const component = shallow(<WorkoutEdit
        toggle={toggle}
        modal={true}
        item={item}
        maxDate={maxDate}
        onChange={onChange}
        onChangeDate={onChangeDate}
        onEdit={onEdit} />);        

        const tree = toJson(component);
        expect(tree).toMatchSnapshot();

        component.find('Button[color="primary"]').simulate('click');
        expect(onEdit).toHaveBeenCalledTimes(1);

        component.find('Button[color="secondary"]').simulate('click');
        expect(toggle).toHaveBeenCalledTimes(1);
});