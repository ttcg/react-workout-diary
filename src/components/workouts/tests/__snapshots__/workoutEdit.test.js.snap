// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={true}
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[MockFunction]}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    toggle={[MockFunction]}
    wrapTag="div"
  >
    Edit Workout
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <FormGroup
        tag="div"
      >
        <Label
          for="Id"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Id
        </Label>
        <Input
          plaintext={true}
          readOnly={true}
          type="text"
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Label
          for="Date"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Date
        </Label>
        <br />
        <DatePicker
          allowSameDay={false}
          className="form-control"
          dateFormat="dd/MM/yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          id="Date"
          maxDate={2019-01-31T00:00:00.000Z}
          monthsShown={1}
          name="date"
          nextMonthButtonLabel="Next month"
          onBlur={[Function]}
          onChange={[MockFunction]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2019-01-02T00:00:00.000Z}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Label
          for="WorkoutType"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Type
        </Label>
        <Input
          id="WorkoutType"
          name="workoutType"
          onChange={[MockFunction]}
          type="select"
        >
          <option>
            Running
          </option>
          <option>
            Cycling
          </option>
        </Input>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <Label
          for="Calories"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Calories
        </Label>
        <Input
          id="Calories"
          name="calories"
          onChange={[MockFunction]}
          placeholder="Calories burnt"
          type="number"
        />
      </FormGroup>
    </Form>
  </ModalBody>
  <ModalFooter
    tag="div"
  >
    <Button
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Save
    </Button>
     
    <Button
      color="secondary"
      onClick={[MockFunction]}
      tag="button"
    >
      Cancel
    </Button>
  </ModalFooter>
</Modal>
`;
